### PR TITLE
assert: fix flaky TestNeverTrue

### PR DESCRIPTION
**Pull request by dolmen**
## Summary
Fix flaky `TestNeverTrue` in `assert` package: use a channel to make a synchronized counter to avoid a race condition.

## Changes
Refactor TestNeverTrue to synchronize the counter.

## Motivation
Fix flaky test which make CI unreliable.
```console
$ go test -race -failfast -count=100 ./...
```

## Related issues
* #1412 A fix for another flaky test

### CODE REVIEW

**Comment by dolmen** [comment]
Cc: @SuperQ @brackendawson for review

**Review by SuperQ — APPROVED** [review]
Way more clear/correct.

File: assert/assertions_test.go [hunk 1]
```diff
@@ -2794,14 +2794,19 @@ func TestNeverFalse(t *testing.T) {
...
+// TestNeverTrue checks Never with a condition that returns true on second call.
 func TestNeverTrue(t *testing.T) {
 	mockT := new(testing.T)
-	state := 0
+
+	// A synchronized counter using a channel
+	counter := make(chan int, 2)
+	counter <- 1
+	counter <- 2
+	defer close(counter)
```
Comment (brackendawson) [inline assert/assertions_test.go:2805]: This channel is absolutely a solid fix. :+1: But I think a chan of bools is just more clear. And you don't need to defer the close().
```suggestion
	returns := make(chan bool, 2)
	returns <- false
	returns <- true
	close(returns)
```

**Review by brackendawson — CHANGES_REQUESTED** [review]

File: assert/assertions_test.go (hunk 1 above)
Comment (dolmen) [reply assert/assertions_test.go:2805]: Oh, yes!
I'm fixing.

Comment (dolmen) [reply assert/assertions_test.go]: Fixed and rebased.

**Comment by dolmen** [comment]
@brackendawson Fixed and rebased. Ready for merge.

**Review by brackendawson — APPROVED** [review]

**Pull request merged**